### DISCUSSION TITLE
Lines will overlap

### DISCUSSION THREAD

**Discussion by limyxi** [bcakmakoglu/vue-flow#286] [Q&A]
![image](https://user-images.githubusercontent.com/20531560/190384712-95425168-df22-4868-906c-aad956a08b46.png)

How to avoid overlapping lines，it will make the flow unclear

**Comment by bcakmakoglu** [comment]
Have you tried using `step` or `smoothstep` edges instead? They take a different path which might be more to your liking.
Bezier edges don't have a routing that prevents overlaps or calculates paths around nodes.

**Reply by limyxi** [reply]
![image](https://user-images.githubusercontent.com/20531560/190386539-a0069094-7a37-46dc-962c-789248007b7a.png)
step will like this.  But i want the flow like this
![image](https://user-images.githubusercontent.com/20531560/190386743-59e58c81-de7e-4234-ba99-96fa4267b311.png)

Is there a better solution? please help~

**Reply by bcakmakoglu** [reply]
What exactly is the issue? You're not making it clear in the second picture.

You say "step will like this but I want it like this" and show me a picture of a graph with step edges. So you want step edges?
Then use the edge type `step` 😄 

Like I said, Bezier edge (the default curved ones) do not stop overlaps - if you need that sort of routing you need to implement your own custom edge for it.

**Reply by limyxi** [reply]
haha... the second picture is what i want ，the red line do not overlaps ，it's a demand picture.  The effect is not achieved by 'vue-flow'.
The first picture is achieved  by 'vue-flow'  using step edges, but it overlaps。 

i want edge do not ovelaps

**Reply by bcakmakoglu** [reply]
The first picture does not show step edges though.
If you use custom edges you need to use the `getSmoothStepPath` utility and use `borderRadius: 0` 
otherwise you need to use `edge.type = 'step'` 

See this example on the docs homepage: https://vueflow.dev/

<img width="734" alt="image" src="https://user-images.githubusercontent.com/78412429/190404040-5f826c01-532f-4fcc-919a-d71b686dfd6a.png">

**Reply by bcakmakoglu** [reply]
You can see the different edge types here as well: https://vueflow.dev/examples/edges/

**Reply by limyxi** [reply]
![image](https://user-images.githubusercontent.com/20531560/190405123-1d308588-4877-47aa-bbb0-b2b9db055a36.png)
The step circled in red  uses step edges。 I want it to connect outside,  Not in the middle

**Reply by Cooolyu** [reply]
@limyxi How did you solve it in the end?
